eng/pipelines/templates/jobs/publish-coverage.yml

```diff
@@ -0,0 +1,69 @@
+parameters:
+  - name: ServiceDirectory
+    type: string
+    default: ''
+  - name: Artifacts
+    type: object
+    default: []
+  - name: DependsOn
+    type: string
+    default: ''
+  - name: RunCoverage
+    type: boolean
+    default: true
+
+
+jobs:
+  - job: Publish_Code_Coverage
+
+    steps:
+      # Download all existing artifacts (most of the time, it's whatever was uploaded before)
+      - pwsh: |
+          New-Item -Path $(Build.SourcesDirectory) -Name "_all_coverage_files" -ItemType "directory"
+        displayName: 'Create all coverages directory'
+        continueOnError: true
+        condition: and(succeededOrFailed(), ${{ parameters.RunCoverage }})
+
+      - task: DownloadPipelineArtifact@2
+        inputs:
+          source: current
+          # patterns: '**' # Download everything
+          path: '$(Build.SourcesDirectory)/_all_coverage_files'
+
+      - pwsh: |
+          pip install coverage==4.5.4
+        displayName: Install coverage
+
+      - pwsh: |
+          Get-ChildItem -Recurse .\_all_coverage_files\ |
+          ForEach-Object {
+            If (Test-Path $_ -PathType Leaf) {
+              Get-Content $_
+            }
+          }
+        displayName: Show all coverage files
+
+      - task: PythonScript@0
+        displayName: 'Create Coverage Report'
+        inputs:
+          scriptPath: 'scripts/devops_tasks/create_coverage.py'
+
+      - pwsh: |
+          Get-Content $(Build.SourcesDirectory)\.coverage
+
+      - pwsh: |
+          Get-Content $(Build.SourcesDirectory)\coverage.xml
+
+      - task: PublishCodeCoverageResults@1
+        displayName: 'Publish Code Coverage to DevOps'
+        continueOnError: true
+        condition: and(succeededOrFailed(), ${{ parameters.RunCoverage }})
+        inputs:
+          codeCoverageTool: Cobertura
+          summaryFileLocation: '$(Build.SourcesDirectory)/coverage.xml'
+
+  - job: Show_all_coverage_files
```

Comment:
Why not do this as another step in the previous job? We don't want to use another agent machine to just do that.